### ISSUE TITLE
“Propose alt text” feature

### ISSUE THREAD

**Issue by nclm** [mastodon/mastodon#27320]
### Pitch

Alt text is crucial but not everyone know about it, remembers about it, or are good at writing them. An interaction I see very often on Mastodon, is people replying to a picture without alt text (or with poor alt text) with a proposition for a relevant alt text. It is nice and helpful that alt text can be crowdsourced like this, but it’s often lost in the replies. The original poster might also not know how to copy that alt text and edit their post media to paste the alt text there.

This is an idea about providing UI for that. The interaction would not fundamentally change, but it would be smoother and enable easy crowdsourced editing of the picture alt text.

How I imagine it:
- When a post does not have alt text, a “no alt / propose alt” badge/button would be displayed. Clicking it would go to the next step. If a post already have alt text, clicking the “alt” badge would open a modal with the alt text nicely readable, along with a “suggest an edit” button which also goes to the next step. Meaning that readers of the post, not only the original poster, have an easy way to write alt text for any images that they come across.
- When someone clicks, it open a new write alt text modal. This can be just based on the modal used when editing your own alt text for an image you post. If there was no alt text on the image, it would be an empty field, if there was already alt text, it would be prefilled with the original.
- Once written/edited, you can click “Submit/Propose”
- The original poster would get a notification (of a new type), “New alt text proposal for this media”, with a simple “Accept” button (and maybe “Edit” too).
- Once accepted, the original post would be automatically edited and republished, with the new alt text inserted. The alt text writer could also be credited in the UI when clicking on the “alt” badge.

In the back-end, alt text proposals doesn’t have to be anything else than normal replies, maybe with a special tag or starting with “Alt text:” (meaning compatibility with final fallback for any other Fediverse software or clients). But it would make suggesting alt texts simpler and more integrated as an expected interaction when alt text is missing/poor.

It seems like a small change (not much new UI, just special replies) that is out of the way when not relevant but would add a lot to this specific interaction?

**Comment by AdeleHorn** [comment]
Generally, that's a nice idea. Except that certain users would probably inundate their contemporaries with looong epistles and edits. Which the original poster would then have to deal with one by one. My guess is, most would quickly disable their notifications for that function, since there are _loads_ of self-anointed alt text sherriffs out there.

**Comment by thibaultmol** [comment]
duplicate of https://github.com/mastodon/mastodon/issues/26464 ?

**Comment by nclm** [comment]
> duplicate of #26464 ?

Oops indeed thanks! Closing this.

**Comment by nclm** [comment]
Duplicate of #26464 26464
